pandas/tests/groupby/test_groupby.py

```diff
@@ -1300,17 +1301,6 @@ def test_non_cython_api(self):
         g = df.groupby('A')
         gni = df.groupby('A', as_index=False)
 
-        # mad
```

Comment:
This test was failing with this change because it allowed for `mad` operations on `object` types, returning `NaN` rather than raising. I figured the explicit tests added elsewhere made more sense and that raising is preferable to `NaN` for particular types.

There's also some code that tests `any` further down within this test. Didn't check in detail but can probably be removed on account of changes done in #19722. Assuming I have updates I'll include in the next batch or alternately open up a separate issue